x-pack/plugins/ml/public/privilege/check_privilege.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+// @ts-ignore
+import { hasLicenseExpired } from '../license/check_license';
+
+import { Privileges } from './common';
+import { getPrivileges } from './get_privileges';
+
+let privileges: Privileges = {};
+
+// Helper type to allow `new Promise(...)`;
+type PromiseConstructor<T> = new (
+  arg: ((resolve: (arg?: T) => void, reject: (arg?: T) => void) => void)
+) => Promise<T>;
+
+// Privileges Promise Constructor
+type PPC = PromiseConstructor<Privileges>;
+// Privileges Promise
+type PP = Promise<Privileges>;
+
+export function checkGetJobsPrivilege(Promise: PPC, kbnUrl: any): PP {
```

Comment:
Promise doesn't need to be passed in here. If I remember correctly, I believe the old version of this function passed in `Promise` because kibana was defining their own `Promise` and there were some issues. It is really old code and we should be using the native `Promise` now.

Reply:
Good point I totally forgot that it's not required anymore. Fixed in b14227d.